### PR TITLE
Complete parTraverse ops for Either and Validated

### PR DESCRIPTION
**Pull request by nomisRev**
## Status
**READY**

## Description
Completes the `parTraverse` operations for `Either` and `Validated`.

## Todos
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Review by raulraja — APPROVED** [review]
thanks @nomisRev !

File: arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/ParTraverseValidated.kt [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  f: suspend (A) -> Validated<E, B>
+): Validated<E, List<B>> {
+  val semaphore = Semaphore(n)
+  return parTraverseValidated(ctx, semigroup) { a ->
+    semaphore.withPermit { f(a) }
+  }
+}
+
+/**
+ * Sequences all tasks in parallel on [Dispatchers.Default] and return the result.
```
Comment (i-walker) [inline arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/ParTraverseValidated.kt:84]:
```suggestion
 * Sequences all tasks in parallel on [Dispatchers.Default] and returns the result.
```

File: arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/ParTraverseValidated.kt [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+ * If one or more of the tasks returns [Validated.Invalid] then all the [Validated.Invalid] results will be combined using [semigroup].
+ *
+ * Cancelling this operation cancels all running tasks.
+ */
+suspend fun <E, A> Iterable<suspend () -> Validated<E, A>>.parSequenceEither(semigroup: Semigroup<E>): Validated<E, List<A>> =
+  parTraverseValidated(Dispatchers.Default, semigroup) { it() }
+
+/**
+ * Sequences all tasks in parallel on [ctx] and return the result.
```
Comment (i-walker) [inline arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/ParTraverseValidated.kt:93]:
```suggestion
 * Sequences all tasks in parallel on [ctx] and returns the result.
```

**Review by i-walker — APPROVED** [review]
Thank you @nomisRev  
🙌 
I left a few Nits

**Pull request merged**